File: backend/server2/db/postgresql/postgresql.go
package postgresql

import (
	"database/sql"
	"fmt"
	"os"
	"server2/db/postgresql/pgquerry"
	"strconv"

	"github.com/joho/godotenv"
	_ "github.com/lib/pq"
)

func CheckError(err error) {
	if err != nil {
		panic(err)
	}
}

func PostgreSQLDatabaseConnect() (*sql.DB, error) {

	err := godotenv.Load()
	if err != nil {
		fmt.Println("error while fetching postgresql db config")
	}

	host := os.Getenv("POSTGRESQL_HOST")
	tempPort := os.Getenv("POSTGRESQL_PORT")
	user := os.Getenv("POSTGRESQL_USER")
	password := os.Getenv("POSTGRESQL_PASSWORD")
	dbname := os.Getenv("POSTGRESQL_DATABASE_NAME")

	port, err := strconv.Atoi(tempPort)
	if err != nil {
		fmt.Println("error while fetching postgresql db config", err)
		return &sql.DB{}, err
	}

	// Connection string
	pgsqlconnection := fmt.Sprintf("host=%s port=%d user=%s password=%s dbname=%s sslmode=disable", host, port, user, password, dbname)

	// Open database connection
	db, err := sql.Open("postgres", pgsqlconnection)
	if err != nil {
		CheckError(err)
		fmt.Println("error while connecting postgresql", err)
		return &sql.DB{}, err
	}

	pgquerry.DeclarePGQuerry(db)

	fmt.Println("postgresql database connected!")

	return db, nil

}

File: backend/server1/src/db/postgresql/postgresql.ts
import { drizzle } from "drizzle-orm/node-postgres";
import { Pool } from "pg";
import { configDotenv } from "dotenv";
import * as schema from "@/db/postgresql/schema";
configDotenv();

if (
    !process.env.POSTGRESQL_HOST ||
    !process.env.POSTGRESQL_PORT ||
    !process.env.POSTGRESQL_USER ||
    !process.env.POSTGRESQL_PASSWORD ||
    !process.env.POSTGRESQL_DATABASE_NAME) {
    console.log("postgresql connection failed");
}
const pool = new Pool({
    host: process.env.POSTGRESQL_HOST,
    port: parseInt(process.env.POSTGRESQL_PORT as string),
    user: process.env.POSTGRESQL_USER,
    password: process.env.POSTGRESQL_PASSWORD,
    database: process.env.POSTGRESQL_DATABASE_NAME,
});

console.log("postgresql database connected");

export const db = drizzle(pool, { schema });





